presto-orc/src/main/java/com/facebook/presto/orc/reader/FloatSelectiveStreamReader.java

```diff
@@ -0,0 +1,401 @@
+import com.facebook.presto.orc.stream.InputStreamSources;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockLease;
+import com.facebook.presto.spi.block.ClosingBlockLease;
+import com.facebook.presto.spi.block.IntArrayBlock;
+import com.facebook.presto.spi.block.RunLengthEncodedBlock;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.orc.metadata.Stream.StreamKind.DATA;
+import static com.facebook.presto.orc.metadata.Stream.StreamKind.PRESENT;
+import static com.facebook.presto.orc.reader.Arrays.ensureCapacity;
+import static com.facebook.presto.orc.stream.MissingInputStreamSource.missingStreamSource;
+import static com.facebook.presto.spi.type.RealType.REAL;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.lang.Float.floatToRawIntBits;
+import static java.util.Objects.requireNonNull;
+
+public class FloatSelectiveStreamReader
+        implements SelectiveStreamReader
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(FloatSelectiveStreamReader.class).instanceSize();
+    private static final Block NULL_BLOCK = REAL.createBlockBuilder(null, 1).appendNull().build();
+
+    private final StreamDescriptor streamDescriptor;
+    private final TupleDomainFilter filter;
+    private final boolean nullsAllowed;
+    private final boolean outputRequired;
+    private final LocalMemoryContext systemMemoryContext;
+    private final boolean nonDeterministicFilter;
+
+    private InputStreamSource<BooleanInputStream> presentStreamSource = missingStreamSource(BooleanInputStream.class);
+    private InputStreamSource<FloatInputStream> dataStreamSource = missingStreamSource(FloatInputStream.class);
+    private BooleanInputStream presentStream;
+    private FloatInputStream dataStream;
+
+    private boolean rowGroupOpen;
+    private int readOffset;
+    private int[] values;
+    private boolean[] nulls;
+    private int[] outputPositions;
+    private int outputPositionCount;
+    private boolean allNulls;
+    private boolean valuesInUse;
+
+    public FloatSelectiveStreamReader(
+            StreamDescriptor streamDescriptor,
+            Optional<TupleDomainFilter> filter,
+            boolean outputRequired,
+            LocalMemoryContext systemMemoryContext)
+    {
+        this.streamDescriptor = requireNonNull(streamDescriptor, "streamDescriptor is null");
+        this.filter = filter.orElse(null);
+        this.outputRequired = outputRequired;
+        this.systemMemoryContext = requireNonNull(systemMemoryContext, "systemMemoryContext is null");
+        this.nonDeterministicFilter = this.filter != null && !this.filter.isDeterministic();
+        this.nullsAllowed = this.filter == null || this.filter.testNull();
```

Comment:
nulls should be allowed if filter is non-deterministic; consider adding a test that uses `array(real)` to test this

`nullsAllowed = this.filter == null || nonDeterministicFilter || this.filter.testNull();`

Reply:
yup, missed that..added a test that it fails in filter.testNull method. put a comment to enable the test after adding support for all scalars in a list.

Reply:
done